tools/devtool

```diff
@@ -536,6 +536,7 @@ cmd_test() {
         --privileged \
```

Comment:
The increase seems a bit too much. Before I recommend to check which process is using more than or close to 1024 file descriptors, check if is valid or are we leaking fds. You can use strace -f -e open,close pytest....
I would be gentle in increasing the limit. Maybe try 2 or 4 times.

Reply:
It doesn't look like we're leaking file descriptors. A scaled down strace output for one test looks like this:

```
11:47:33 open("/proc/self/fd", O_RDONLY|O_CLOEXEC|O_DIRECTORY) = 3
11:47:33 close(3)                       = 0
11:47:33 close(3)                       = -1 EBADF (Bad file descriptor)
11:47:33 open("/firecracker/build/cargo_target/x86_64-unknown-linux-musl/release/firecracker", O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_PATH) = 3
11:47:33 close(3)                       = 0
11:47:33 open("/srv/jailer", O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_PATH) = 3
11:47:33 close(3)                       = 0
11:47:33 open("/firecracker/build/cargo_target/x86_64-unknown-linux-musl/release/firecracker", O_RDONLY|O_CLOEXEC) = 3
11:47:33 open("/srv/jailer/firecracker/fa5f5c92-b62e-41ce-b784-773ebd631b20/root/firecracker", O_WRONLY|O_CREAT|O_TRUNC|O_CLOEXEC, 0100755) = 4
11:47:33 close(4)                       = 0
11:47:33 close(3)                       = 0
11:47:33 open("/var/run/netns/fa5f5c92-b62e-41ce-b784-773ebd631b20", O_RDONLY|O_CLOEXEC) = 3
11:47:33 close(3)                       = 0
11:47:33 open("/proc/mounts", O_RDONLY|O_CLOEXEC) = 3
11:47:33 close(3)                       = 0
11:47:33 open("/sys/fs/cgroup/cpuset/firecracker/cpuset.cpus", O_RDONLY|O_CLOEXEC) = 3
11:47:33 close(3)                       = 0
11:47:33 open("/sys/fs/cgroup/cpuset/firecracker/fa5f5c92-b62e-41ce-b784-773ebd631b20/cpuset.cpus", O_WRONLY|O_CREAT|O_TRUNC|O_CLOEXEC, 0666) = 3
11:47:33 close(3)                       = 0
11:47:33 open("/sys/fs/cgroup/cpuset/firecracker/cpuset.mems", O_RDONLY|O_CLOEXEC) = 3
11:47:33 close(3)                       = 0
11:47:33 open("/sys/fs/cgroup/cpuset/firecracker/fa5f5c92-b62e-41ce-b784-773ebd631b20/cpuset.mems", O_WRONLY|O_CREAT|O_TRUNC|O_CLOEXEC, 0666) = 3
11:47:33 close(3)                       = 0
11:47:33 open("/sys/fs/cgroup/cpuset/firecracker/fa5f5c92-b62e-41ce-b784-773ebd631b20/cpuset.mems", O_WRONLY|O_CREAT|O_TRUNC|O_CLOEXEC, 0666) = 3
11:47:33 close(3)                       = 0
11:47:33 open("/sys/devices/system/node/node0/cpulist", O_RDONLY|O_CLOEXEC) = 3
11:47:33 close(3)                       = 0
11:47:33 open("/sys/fs/cgroup/cpuset/firecracker/fa5f5c92-b62e-41ce-b784-773ebd631b20/cpuset.cpus", O_WRONLY|O_CREAT|O_TRUNC|O_CLOEXEC, 0666) = 3
11:47:33 close(3)                       = 0
11:47:33 open("/sys/fs/cgroup/cpuset/firecracker/fa5f5c92-b62e-41ce-b784-773ebd631b20/tasks", O_WRONLY|O_CREAT|O_TRUNC|O_CLOEXEC, 0666) = 3
11:47:33 close(3)                       = 0
11:47:33 open("/sys/fs/cgroup/cpu,cpuacct/firecracker/fa5f5c92-b62e-41ce-b784-773ebd631b20/tasks", O_WRONLY|O_CREAT|O_TRUNC|O_CLOEXEC, 0666) = 3
11:47:33 close(3)                       = 0
11:47:33 open("/sys/fs/cgroup/pids/firecracker/fa5f5c92-b62e-41ce-b784-773ebd631b20/tasks", O_WRONLY|O_CREAT|O_TRUNC|O_CLOEXEC, 0666) = 3
11:47:33 close(3)                       = 0
11:47:33 open("/dev/null", O_RDWR)      = 3
11:47:33 close(3)                       = 0
11:47:33 open("log_fifo", O_RDWR|O_NONBLOCK|O_CLOEXEC) = 3
11:47:33 open("/vmlinux.bin", O_RDONLY|O_CLOEXEC) = 12
11:47:33 open("/xenial.rootfs.ext4", O_RDWR|O_CLOEXEC) = 14
11:47:33 open("/dev/net/tun", O_RDWR|O_NONBLOCK|O_CLOEXEC) = 20
11:47:33 close(28)                      = 0
11:47:33 open("/dev/kvm", O_RDWR|O_CLOEXEC) = 28
11:47:33 close(28)                      = 0
11:47:36 +++ killed by SIGKILL +++
```

There is one point where, when the VM is launched there multiple files are opened/closed, but we should investigate this further by:
1. Running strace as you suggested, but without filtering syscalls, because there are multiple ways of creating file descriptors (such as, openat, dup2).
2. Merging the logs with `strage-log-merge`
3. Tracking the opened/closed file descriptors.

The analysis above could give us an insight on how many fds are opened and where that peak happens.

Regarding this PR, we can try with 4096 and see if it still reproduces.